src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -41,6 +42,10 @@
 
     private final Logger log = LoggerFactory.getLogger(FileService.class);
 
+    private final Map<Path, ScheduledFuture<?>> futures = new HashMap<>();
```

Comment:
Codacy found an issue: [If you run in Java5 or newer and have concurrent access, you should use the ConcurrentHashMap implementation](https://app.codacy.com/gh/ls1intum/Artemis/pullRequest?prid=5968852)